tests/Resources/Page/MarketplaceConsole/MarketplaceConsole.robot

```diff
@@ -0,0 +1,67 @@
+*** Settings ***
+Library         SeleniumLibrary
+Library         Collections
+
+*** Keywords ***
+Is SSO Login Page Visible
+  ${login_prompt_visible} =  Run Keyword and Return Status  Page Should Contain    Enter your email to log in
+  [Return]  ${login_prompt_visible}
+
+Get the OC commands For Cluster registration
+    Launch Cluster Tab from Marketplace
+    sleep    5
+    Wait until Element is Visible          //span[contains(text(),'Generate Secret')]        timeout=20
+    Click Button    Generate Secret
+    Wait Until Keyword Succeeds    30  1   Element Should Be Visible     //button[contains(@class, 'disabled')]/span[contains(text(),'Generate Secret')]
+    sleep  5
+    ${elem} =   Get WebElements      xpath://div[@role='textbox']//pre
+    &{List}   	Create Dictionary
+    @{itemname}   Create List     Create namespace         Red Hat Marketplace Subscription          Red Hat Marketplace Kubernetes Secret    Red Hat Marketplace global pull secret
+    FOR  ${idx}  ${ext_link}  IN ENUMERATE  @{elem}   start=0
+
+        Set To Dictionary       ${List}      ${itemname[${idx}]}    ${ext_link.text}
+
+    END
+    [Return]    ${List}
+
+#Launch Marketplace
+#   Open Browser  ${MARKETPLACE_TEST.URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
+
+Wait For Marketplace Splash Page
+   Wait Until Page Contains Element    xpath://a[contains(text(), 'Workspace')]   timeout=15
+   Page Should Contain      Red Hat Marketplace
+   Sleep    3
+
+Login to Marketplace
+  [Arguments]  ${username}  ${password}
+  Wait Until Element is Visible            //a[contains(@aria-label,"Log in")]
```

Comment:
is it necessary? even though there is `Is SSO Login Page Visible` at line 40?

Reply:
it's not mandatory but I think it is good to have it here , since sometime due to loading speed(internet) some part of page are loaded and some aren't

Reply:
mmmh maybe you can integrate a timeout inside the `Is SSO Login Page Visible` keyword.

Reply:
IMHO not sure it will be good idea to have timeout in "Is SSO Login Page Visible".In case in future is we can Leverage the SSO keyword with other page. It might not be good idea to have wait there if the page has Signup or something instead of Login. Though if you like,I can change the code

Reply:
Done!!